src/payload/rtr.rs

```diff
@@ -75,13 +69,44 @@ pub struct Ipv6Prefix {
 ///
 /// There is at least one more payload type – BGPSEC router keys – that is not
 /// currently supported by the crate.
-#[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+#[derive(Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+#[non_exhaustive]
 pub enum Payload {
-    /// An IPv4 route origin authorisation.
-    V4(Ipv4Prefix),
+    /// A route origin authorisation.
+    Origin(PrefixOrigin),
+
+    /// A BGPsec router key.
+    RouterKey(RouterKey),
+}
 
-    /// An IPv6 route origin authorisation.
-    V6(Ipv6Prefix)
+impl Payload {
+    /// Creates a new prefix origin payload.
+    pub fn origin(prefix: MaxLenPrefix, asn: AsId) -> Self {
+        Payload::Origin(PrefixOrigin::new(prefix, asn))
+    }
+
+    /// Creates a new router key payload.
+    pub fn router_key(
+        key_identifier: KeyIdentifier, asn: AsId, key_info: Bytes
+    ) -> Self {
+        Payload::RouterKey(RouterKey::new(key_identifier, asn, key_info))
+    }
+
+    /// Returns the origin prefix if the value is of the origin variant.
+    pub fn to_origin(&self) -> Option<PrefixOrigin> {
```

Comment:
Yes. By convention, the three prefixes `as_`, `to_`, and `into_` each have a specific implication when used in conversion methods. A method starting with `as_` converts a reference to a value into a reference to something else (think `as_bytes` for `str` which turns a `&str` into a `&[u8]`). Something starting with `to_` converts a reference into an owned value – here `&self` is turned into an `Option<PrefixOrigin>`. (The distinction between the two is important because of lifetimes.) Finally, `into_` actually converts an owned value into something else (such as `Arc::into_inner` after which the arc is gone).